client/build.gradle

```diff
@@ -1,30 +1,32 @@
+import org.springframework.boot.gradle.plugin.SpringBootPlugin
+
 plugins {
     id 'java-library'
+    id 'pmd'
     id 'maven-publish'
 
     id 'com.jfrog.artifactory' version '4.18.2'
-    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
-    id 'org.hidetake.swagger.generator' version '2.18.2'
-    id 'org.springframework.boot' version '2.5.1'
+    id 'io.spring.dependency-management' version '1.1.5'
+    id 'org.hidetake.swagger.generator' version '2.19.2'
+    id 'org.springframework.boot' version '3.2.5'
 }
 
 repositories {
     mavenCentral()
     maven {
-        url "https://broadinstitute.jfrog.io/broadinstitute/libs-release/"
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-release/'
     }
     maven {
-        url "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/"
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/'
     }
     google()
 }
 
-group = "bio.terra"
+group = 'bio.terra'
 version = gradle.tanagraVersion
 
-//// Keep the client library at Java 11 for now
-sourceCompatibility = JavaVersion.VERSION_11
-targetCompatibility = JavaVersion.VERSION_11
+sourceCompatibility = JavaVersion.VERSION_17
+targetCompatibility = JavaVersion.VERSION_17
```

Comment:
What are the implications for services that depend on our client library? Does this mean they also need to be on Java 17? I'm not certain, but we may have both a Java 11 (VUMC admin service) and Java 17 (AoU core workbench service) that depend on this client lib.

Reply:
From my understanding - yes, the service should ALSO switch to java17 to use clients built with targetCompatibility 17. 
A java 17 service can use either java 11 or java 17 (example: this repo uses java 11 admin client) as long as the interfaces are compatible. 
However targetCompatibility 17 will require VUMC to switch other services to java 17 (to consume changes committed to this client after java 17 upgrade). to prevent this dependency I will retain the targetCompatibility 11 and create a followup ticket to change it. ([BENCH-3210](https://verily.atlassian.net/browse/BENCH-3210))

more info - https://stefma.medium.com/sourcecompatibility-targetcompatibility-and-jvm-toolchains-in-gradle-explained-d2c17c8cff7c

Reply:
Moving only target to java 11 did not work as expected. Hence reverted client changes and will be handled separately ([BENCH-3210](https://verily.atlassian.net/browse/BENCH-3210), draft changes https://github.com/DataBiosphere/tanagra/pull/875)
tested local publish

```
tanagra (dexamundsen/build-17) >> ./gradlew client:publishToMavenLocal
...
BUILD SUCCESSFUL in 975ms
16 actionable tasks: 3 executed, 13 up-to-date
tanagra (dexamundsen/build-17) >> ls -l ~/.m2/repository/bio/terra/tanagra-client/
total 8
drwxr-xr-x  7 dexamundsen  primarygroup  224 Jun  8 19:27 0.0.512-SNAPSHOT-LOCAL
-rw-r--r--  1 dexamundsen  primarygroup  381 Jun  8 19:27 maven-metadata-local.xml
```